pkg/specgen/generate/kube/kube.go

```diff
@@ -370,6 +371,10 @@ func ToSpecGen(ctx context.Context, opts *CtrSpecGenOptions) (*specgen.SpecGener
 		}
 
 		volume.MountPath = dest
+		path := volumeSource.Source
+		if len(volume.SubPath) > 0 {
+			path = filepath.Join(path, volume.SubPath)
```

Comment:
This can be dangerous.

What happens if `subpath` contains a symlink?  It can point outside the volume directory.

Reply:
@giuseppe I am looking to take this over and get it done. I agree that this can be dangerous if it is a a symlink that can point outside the volume directory, I was thinking we can resolve the path given and if it is a symlink we return an error. Basically, the subpath cannot be a symlink. Do you think that could work?

Reply:
No that is not enough.  I have recently fixed a similar problem in Podman with subpaths, could you please find it in the history (approximately 1 month ago)?  I can point you to the right commit once I am back at the computer :)

Reply:
I think I found it https://github.com/containers/podman/pull/17992. I will follow that for subpaths here. Thank you!